### PR TITLE
refactor logic for chroma init

### PR DESCRIPTION
**Pull request by jeffchuber**
This PR simplifies the init logic and hopefully makes it reason about. 

For discussion!

- Forces users to pass in 1 of the following: client, client_settings or persist_directory. 
- Gives the user helpful error messages if they dont do this 
- keeps backwards compatibility for pre `0.3.29` however gently encourages users to upgrade

@baskaryan would like your feedback on this :) 

Also - will make sure tests pass...

### CODE REVIEW

**Comment by jeffchuber** [comment]
Ill fix this up tomorrow

**Comment by baskaryan** [comment]
@jeffchuber thoughts on adding some unit tests to make sure things are being instantiated correctly?

**Comment by jeffchuber** [comment]
@baskaryan good idea, i will do that

File: libs/langchain/tests/integration_tests/vectorstores/test_chroma.py [hunk 1]
```diff
@@ -230,23 +241,27 @@ def test_chroma_update_document() -> None:
...
-        metadatas=metadatas,
-        collection_metadata={"hnsw:space": "l2"},
-    )
-    output = docsearch.similarity_search_with_relevance_scores("foo", k=3)
-    assert output == [
-        (Document(page_content="foo", metadata={"page": "0"}), 1.0),
-        (Document(page_content="bar", metadata={"page": "1"}), 0.8),
-        (Document(page_content="baz", metadata={"page": "2"}), 0.5),
-    ]
+# _euclidean_relevance_score_fn is not correct math
```
Comment (jeffchuber) [inline libs/langchain/tests/integration_tests/vectorstores/test_chroma.py:244]: not sure if we want to address this here... this test never passed afaik, and I'm not sure how/why it was included.

**Comment by leo-gan** [comment]
@jeffchuber Hi , could you, please, resolve the merging issues? After that ping me and I push this PR for the review. Thanks!

**Comment by jeffchuber** [comment]
@leo-gan this would still be good to do, i think it majorly cleans up the logic

**Comment by efriis** [comment]
Hey @jeffchuber ! I think this one has started to diverge from the current state of the repo pretty substantially (esp with multimodal this week!). I'll probably close this one for now, and agreed this would be useful to refactor at some point.

If you get that done, feel free to reopen!

**Comment by efriis** [comment]
Actually - looked some more. I'll try to resolve some of these now.

**Pull request closed without merging**